src/django/api/migrations/_tigerline_helper.py

```
@@ -53,6 +53,14 @@ def download_csv_from_s3(s3_key):
         )
 
     s3_client = get_s3_client()
+
+    try:
+        s3_client.head_object(Bucket=bucket_name, Key=s3_key)
+    except s3_client.exceptions.NoSuchKey:
+        raise ValueError(f'CSV file not found: {s3_key}')
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)